common/djangoapps/course_modes/signals.py

```diff
@@ -35,3 +46,47 @@ def _listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable
 def _should_update_date(verified_mode):
     """ Returns whether or not the verified mode should be updated. """
     return not(verified_mode is None or verified_mode.expiration_datetime_is_explicit)
+
+
+def update_masters_access(item, verified_id, masters_id):
+    """
+    Update the XBlock's group access to allow the master's group,
+    in addition to the verified content group.
+    """
+    group_access = item.group_access
+    enrollment_groups = group_access.get(ENROLLMENT_TRACK_PARTITION_ID, None)
+    if enrollment_groups is not None:
+        if verified_id in enrollment_groups and masters_id not in enrollment_groups:
+            enrollment_groups.append(masters_id)
+            item.group_access = group_access
+            return True
+
+
+@receiver(post_save, sender=CourseMode)
+def update_masters_access_course(sender, instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    Update all blocks in the verified content group to include the master's content group
+    """
+    if instance.mode_slug != CourseMode.MASTERS:
+        return
+    masters_id = getattr(settings, 'COURSE_ENROLLMENT_MODES', {}).get('masters', {}).get('id', None)
+    verified_id = getattr(settings, 'COURSE_ENROLLMENT_MODES', {}).get('verified', {}).get('id', None)
+    if not (masters_id and verified_id):
+        log.error("Missing settings.COURSE_ENROLLMENT_MODES -> verified:%s masters:%s", verified, masters)
+        return
+
+    course_id = instance.course_id
+    user = get_current_user()
+    user_id = user.id if user else None
+    store = modulestore()
+
+    with store.bulk_operations(course_id):
+        try:
+            items = store.get_items(course_id, settings={'group_access': {'$exists': True}}, include_orphans=False)
+        except ItemNotFoundError:
+            return
+        for item in items:
+            if update_masters_access(item, verified_id, masters_id):
```

Comment:
What if `update_masters_access` were to become `should_update_masters_access` and returned a boolean representing whether or not to add Master's access to the item, and then the act of adding Master's access were to happen here based on the value of that function? It was unintuitive for me to have the function in the predicate have a side effect. And that way `should_update_masters_access` is idempotent.

Reply:
I removed that separate function. If it's necessary to update access for an individual xblock, someone else can refactor in a better way.